### PR TITLE
Whitelist/Blacklist

### LINKED ISSUE #8: Ability to whitelist items
It would be nice if it were possible to whitelist certain items, like the Digsite pendant or other untradable items so that they can be alched even if the setting to hide untradable items is enabled.

### PR DESCRIPTION
**Pull request by Dragonkiller93**
## What does your PR do?
Adds a whitelist and a blacklist which ignore other rules. 
Blacklist takes priority, so if an item is in both, it won't be shown when alching.
Fixes #8 

## Testing
I tested it a good amount, with different items/untradeables. More testing is always welcome, I may have missed some edge cases.

### CODE REVIEW

**Comment by CreativeTechGuy** [comment]
Thanks for the feature! There's a few things that should be addressed:

* Can you change the verbiage from whitelist/blacklist to allowlist/denylist
* Add to the repo's README an explanation of these new options and examples on how to configure them
* See if there's any examples in other plugins for how to configure lists of items. I feel like trying to type the name in as a CSV (which isn't explained anywhere currently in-game) is very error-prone. A capitalization issue or punctuation issue would cause it to not match. I wonder if we could do something where you can toggle on a setting in the plugin which enables you to right click on items and in the context menu of that item, only while the mode is on, the user could click to add an item to the allow/deny lists. Then we could store the item IDs which are more reliable and the user wouldn't ever need to interact with it.

Other code-level comments will be made in-line.

**Review by CreativeTechGuy — CHANGES_REQUESTED** [review]

File: src/main/java/com/creativetechguy/NoBadAlchsPlugin.java [hunk 1]
```diff
@@ -50,22 +51,35 @@ public class NoBadAlchsPlugin extends Plugin {
             ItemID.EMERALD_6896,
             ItemID.RUNE_LONGSWORD_6897
     );
+    private List<String> allowList = new CopyOnWriteArrayList<>();
```
Comment (Dragonkiller93) [inline src/main/java/com/creativetechguy/NoBadAlchsPlugin.java:54]: I don't know whether this is required or not, I saw it used somewhere else so I figured better safe than sorry. If  unnecessary it can be removed.

**Comment by Dragonkiller93** [comment]
That PR has all the fixes, but not the context menu addition. That might take a bit longer to figure out.

**Comment by Dragonkiller93** [comment]
That's actually exactly what this plugin does: [Alch blocker](https://github.com/robrichardson13/alch-blocker). Maybe a merge should be considered, since both serve the same purpose.

**Comment by CreativeTechGuy** [comment]
Thank you for all of your help here. Let's see how it turns out with that other plugin and what the next steps should be. I'll revisit this next week.

**Comment by Dragonkiller93** [comment]
Whew it's been a minute. The author of that plugin never responded, and there haven't been any updates since Jan 2024. I could just start merging anyways, and see if runelite wants to remove the other plugin since it'll be unnecessary.

**Review by CreativeTechGuy — APPROVED** [review]
Looks good. Thank you!

**Comment by CreativeTechGuy** [comment]
You can track the release here: https://github.com/runelite/plugin-hub/pull/8935
